tests/src/test/scala/akka/kafka/scaladsl/ReconnectSpec.scala

```diff
@@ -195,13 +135,14 @@ class ReconnectSpec extends SpecBase with TestcontainersKafkaPerClassLike {
 
       // shut down
       Await.ready(control.shutdown(), remainingOrDefault)
+      Await.ready(proxyBinding2.unbind(), remainingOrDefault)
     }
   }
 
   /**
    * Create a proxy so it can be shut down.
    */
-  def createProxy(): (Future[Tcp.ServerBinding], Future[UniqueKillSwitch]) = {
+  def createProxy(): (Tcp.ServerBinding, Future[UniqueKillSwitch]) = {
```

Comment:
I wonder if this really works. The Kafka broker will distribute all the broker addresses to the client anyway, so the client will talk to the non-proxied port once it is connected.

Reply:
Good point. We'll need to think of some other way to hangup the connection using the container to preserve these tests. The problem with stopping and starting the container is that state is blown away.

Reply:
I tried a few things today:

* I'm able to pause and unpause the broker containers in place of embedded-kafka stop and start. This doesn't appear to kill any active connections. When I enable Kafka `NetworkClient` logs I can see with the previous embedded-kafka impl. a disconnect would occur.
* I can disconnect and reconnect the network to broker containers. This will force any active connections to close, but when the network is reattached the host port is different. Testcontainers will always generate a random host port and there doesn't appear to be a way around this. https://www.testcontainers.org/features/networking/

Some other ideas

* Use something like iptables to disable connections in the running container itself, but I can't find any of the usual programs in the image to do this (`iptables`, `route`) or even network interface commands.
* Stop and start the broker process within the running container. This should do the trick, but it will require some work to update the container run command to support it.

Reply:
Create an issue about this instead and point to it from the test.

Reply:
https://github.com/akka/alpakka-kafka/issues/1233